nirvana.go

```diff
@@ -169,13 +169,9 @@ func NewDefaultConfig() *Config {
 // modifiers for specific scenario, please use NewDefaultConfig().
 func NewConfig() *Config {
 	return &Config{
-		ip:          "",
-		port:        8080,
-		logger:      &log.SilentLogger{},
-		filters:     []service.Filter{},
-		descriptors: []definition.Descriptor{},
-		modifiers:   []service.DefinitionModifier{},
-		configSet:   make(map[string]interface{}),
+		port:      8080,
+		logger:    &log.SilentLogger{},
+		configSet: make(map[string]interface{}),
```

Comment:
Why do you remove these field initializers?

Reply:
Because the values assigned to them are default ones, which is not necessary to explicitly write.
Besides, nil slice can append items, while using nil maps will cause panic.

Reply:
It's a convention and not a constraint. I recommend initialize everything in the constructor.

Reply:
It sounds reasonable, but what if there are too many fields, like 10 or more?

Reply:
It's not a usual case. We always have a lot of fields to initialize.

Reply:
For simplicity, I prefer to only write fields whose assigned values are not default ones.
Or is there a clear line between structs that all the fields should be explicitly initialized and those not?

Reply:
do we agree upon on this change?